### PR TITLE
Packages

### PR DESCRIPTION
**Pull request by sklee1216**

### CODE REVIEW

**Comment by realRoshanRaj** [comment]
Whys the linter failing?

**Comment by sklee1216** [comment]
> Whys the linter failing?

I have no idea. we need to update @actions/core but idk how to do.

File: pkg/db/firebase.go [hunk 1]
```diff
@@ -274,3 +346,47 @@ func recordActionEntry(client *firestore.Client, ctx context.Context, action str
...
+			}
+		} else if mode == "Tilde" {
+			tilde, _ := semver.NewConstraint(version)
+			pkgVersion, _ := semver.NewVersion(pkg.Metadata.Version)
+
+			if pkg.Metadata.Name == name && tilde.Check(pkgVersion) {
+				tmp := models.PackageQuery{Name: pkg.Metadata.Name, Version: pkg.Metadata.Version}
+				pkgs = append(pkgs, tmp)
+			}
+		}
```
Comment (realRoshanRaj) [inline pkg/db/firebase.go]: Carat and Tilde have the same logic, no?

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -81,15 +84,220 @@ func extractPackageJsonFromZip(encodedZip string) (*PackageJson, bool) {
...
+		repourl = str
+		// fmt.Println(str)
+	} else if repo, ok := pkgJson.Repository.(RepoPackageJson); ok {
+		repourl = repo.URL
+		// fmt.Println(repo.URL)
+	} else if m, ok := pkgJson.Repository.(map[string]interface{}); ok {
+		if url, ok := m["url"].(string); ok {
+			repourl = url
+			// fmt.Println(url)
+		}
```
Comment (realRoshanRaj) [inline pkg/utils/utils.go]: Need to properly parse and store the repository url in the same form

**Review by realRoshanRaj — CHANGES_REQUESTED** [review]

**Review by realRoshanRaj — APPROVED** [review]

**Pull request merged**